internal/resourceproviders/requiring_registration.go

```diff
@@ -20,8 +23,13 @@ func DetermineWhichRequiredResourceProvidersRequireRegistration(requiredResource
 		}
 
 		if _, isUnregistered := (*unregisteredResourceProviders)[providerName]; !isUnregistered {
-			// this is likely a typo in the Required Resource Providers list, so we should surface this
-			return nil, fmt.Errorf("the required Resource Provider %q wasn't returned from the Azure API", providerName)
+			if environment.Name == environments.AzurePublicCloud {
+				// this is likely a typo in the Required Resource Providers list, so we should surface this
+				return nil, fmt.Errorf("the required Resource Provider %q wasn't returned from the Azure API", providerName)
+			}
```

Comment:
thinking about this we should probably remove this error here, and just log all the time - the unit test should fail but if an RP gets removed we shouldn't flag an error here?